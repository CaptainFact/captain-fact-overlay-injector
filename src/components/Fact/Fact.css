.fact {
  display: block;
  border: 1px solid rgba(101, 101, 101, 0.65)
}

.fact:not(:last-child) {
  margin-bottom: 0.25em;
}

.sourceSection {
  min-height: 2em;
  padding: 0.1em;
}

.scoreTag {
  font-size: 0.9em;
  font-family: sans-serif;
  font-weight: bold;
  color: #cec8c8;
  background: rgba(53, 53, 53, 0.5);
  padding: 0.4em 0.5em;
  margin: 0.2em;
  border-radius: 4px;
  float: right;
}

.scoreTag span, .scoreTag img {
  vertical-align: middle;
}

.scoreTag img {
  height: 0.8em;
}

.userSection {
  font-size: 0.8em;
  padding: 5px 10px;
  background-color: #4a4343;
  display: flex;
}

.userSection > .right {
  padding-left: 7px;
}

.userCommentText {
  font-style: italic;
  color: #bbbbbb;
  text-shadow: 0 0 1px rgba(0, 0, 0, 0.45);
}

/* Approving facts */

.approvingFact > .sourceSection {
  background-color: #2EBE00;
}

.approvingFact > .sourceSection:hover {
  background-color: #2bb300;
}

.approvingFact > .userSection {
  background-color: #228c00;
}

/* Refuting facts */

.refutingFact > .sourceSection {
  background-color: #c5132c;
}

.refutingFact > .sourceSection:hover {
  background-color: #bd122a;
}

.refutingFact > .userSection {
  background-color: #8a2b38;
}

/* Regular fact */
.regularFact > .sourceSection {
  background: #636363;
}
