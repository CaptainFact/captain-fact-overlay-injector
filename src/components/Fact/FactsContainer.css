.container {
  padding: 0.25em;
}