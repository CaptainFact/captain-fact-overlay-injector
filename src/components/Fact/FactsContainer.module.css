.title {
  background: #f1f1f1;
  border-bottom: 1px solid #dbdbdb;
  color: #4a4a4a;
  padding: 0.25em;
  text-align: center;
  box-shadow: 0px -5px 5px #2e4b5d;
}
