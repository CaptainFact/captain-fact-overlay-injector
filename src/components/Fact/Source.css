.sourceLink {
  text-decoration-color: lightgrey;
  display: block;
  text-shadow: 0 0 1px rgba(0, 0, 0, 0.15);
  text-decoration: none;
  padding: 0.5em;
}

.sourceLink img {
  height: 1em;
}

.sourceLink:hover .articleTitle {
  text-decoration: underline;
}

.siteName {
  text-decoration: underline;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: white;
  font-weight: bold;
  font-size: 0.9em;
  vertical-align: top;
}

.articleTitle {
  font-size: 0.9em;
  color: #e0e0e0;
  text-decoration-color: lightgrey;
}
