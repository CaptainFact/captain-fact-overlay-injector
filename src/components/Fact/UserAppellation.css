.container > * {
    text-decoration: underline;
    color: #0095ff;
}