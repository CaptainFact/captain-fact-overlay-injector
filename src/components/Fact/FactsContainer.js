import React from 'react'

import Fact from './Fact'

import { title } from './FactsContainer.css'


export default class FactsContainer extends React.PureComponent {
  render() {
    return (
      <div>
        <div className={title}>Sources</div>
        {this.props.comments.map(comment =>
          <Fact key={comment.id} comment={comment}/>
        ).toArray()}
      </div>
    )
  }
}
