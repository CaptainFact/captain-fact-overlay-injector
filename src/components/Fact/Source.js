import React from 'react'
import upperCase from 'voca/upper_case'

import { siteNameContainer, siteName, articleTitle, sourceLink } from './Source.css'
import DEFAULT_IMG_NEW_TAB from '../../assets/new_tab.png'


const getHostName = url =>
  upperCase(url.replace(/https?:\/\//, '').replace(/\/.*/g, ''))

const Source = ({source: {url, title, site_name}, imgNewTab = DEFAULT_IMG_NEW_TAB}) => (
  <a href={url} target="_BLANK" className={sourceLink}>
    <div className={siteNameContainer}>
      <img src={imgNewTab}/>&nbsp;
      <span className={siteName}>
        { upperCase(site_name) || getHostName(url) }
      </span>
    </div>
    <span className={articleTitle}>{title}</span>
  </a>
)

export default Source
