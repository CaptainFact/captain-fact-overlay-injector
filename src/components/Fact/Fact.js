import React from 'react'
import Source from './Source.js'
import UserAppellation from './UserAppellation'
import { fact, approvingFact, refutingFact, regularFact, userSection, right, scoreTag, sourceSection, userCommentText } from './Fact.css'
import { staticResource } from '../../lib/static_resource'
import starImageFile from '../../assets/star.png'


const getUserPicture = (userId, url, url_mini, size) => {
  if (!url || !url_mini)
    return `https://api.adorable.io/avatars/${size}/${userId}.png`
  return size <= 48 ? staticResource(url_mini) : staticResource(url)
}

export default class Fact extends React.PureComponent {
  render() {
    const { approve, text, source, user, score } = this.props.comment
    const factType = approve === true ? approvingFact : approve === false ? refutingFact : regularFact

    return (
      <div className={`${fact} ${factType}`}>
        <div className={sourceSection}>
          <span className={scoreTag}>
            <span>{score || 0} </span>
            <img src={this.props.imgStar || starImageFile}/>
          </span>
          <Source source={source}/>
        </div>
        {text && text.length > 0 &&
          <div className={userSection}>
            <img src={getUserPicture(user.id, user.picture_url, user.mini_picture_url, 32)} height="28"/>
            <div className={right}>
              <UserAppellation user={user}/>&nbsp;
              <span className={userCommentText}>{text}</span>
            </div>
          </div>
        }
      </div>
    )
  }
}

