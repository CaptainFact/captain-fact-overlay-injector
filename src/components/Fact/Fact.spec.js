import Fact from './Fact'


test("approving fact", () => {
  snapshot(<Fact comment={{user: {id: 42}, approve: true, score: 1337}}/>)
})

test("refuting fact", () => {
  snapshot(<Fact comment={{user: {id: 42}, approve: false, score: 1337}}/>)
})

test("with text", () => {
  snapshot(<Fact comment={{user: {id: 42}, approve: true, score: 1337, text: "Foobar"}}/>)
})

test("custom star image", () => {
  snapshot(<Fact comment={{user: {id: 42}, approve: true, score: 1337}} imgStar="custom.jpg"/>)
})