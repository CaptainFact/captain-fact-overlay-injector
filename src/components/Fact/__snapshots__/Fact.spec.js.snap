// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`approving fact 1`] = `
<div
  className="fact approvingFact"
>
  <div
    className="sourceSection"
  >
    <span
      className="scoreTag"
    >
      <span>
        1337
         
      </span>
      <img
        src="test-file-stub"
      />
    </span>
    <Source />
  </div>
</div>
`;

exports[`custom star image 1`] = `
<div
  className="fact approvingFact"
>
  <div
    className="sourceSection"
  >
    <span
      className="scoreTag"
    >
      <span>
        1337
         
      </span>
      <img
        src="custom.jpg"
      />
    </span>
    <Source />
  </div>
</div>
`;

exports[`refuting fact 1`] = `
<div
  className="fact refutingFact"
>
  <div
    className="sourceSection"
  >
    <span
      className="scoreTag"
    >
      <span>
        1337
         
      </span>
      <img
        src="test-file-stub"
      />
    </span>
    <Source />
  </div>
</div>
`;

exports[`with text 1`] = `
<div
  className="fact approvingFact"
>
  <div
    className="sourceSection"
  >
    <span
      className="scoreTag"
    >
      <span>
        1337
         
      </span>
      <img
        src="test-file-stub"
      />
    </span>
    <Source />
  </div>
  <div
    className="userSection"
  >
    <img
      height="28"
      src="https://api.adorable.io/avatars/32/42.png"
    />
    <div
      className="right"
    >
      <UserAppellation
        user={
          Object {
            "id": 42,
          }
        }
      />
      Â 
      <span
        className="userCommentText"
      >
        Foobar
      </span>
    </div>
  </div>
</div>
`;
